shared/src/simulation/activities/recurring-event.ts

```diff
@@ -0,0 +1,72 @@
+import { Type } from 'class-transformer';
+import { IsInt, IsUUID, Min, ValidateNested } from 'class-validator';
+import { cloneDeep } from 'lodash-es';
+import { getCreate } from '../../models/utils';
+import { UUID, uuidValidationOptions } from '../../utils';
+import { IsValue } from '../../utils/validators';
+import { ExerciseSimulationEvent, simulationEventTypeOptions } from '../events';
+import { sendSimulationEvent } from '../events/utils';
+import type {
+    SimulationActivity,
+    SimulationActivityState,
+} from './simulation-activity';
+
+export class RecurringEventActivityState implements SimulationActivityState {
+    @IsValue('recurringEventActivity' as const)
+    public readonly type = 'recurringEventActivity';
+
+    @IsUUID(4, uuidValidationOptions)
+    public readonly id: UUID;
+
+    @Type(...simulationEventTypeOptions)
+    @ValidateNested()
+    public readonly event: ExerciseSimulationEvent;
+
+    @IsInt()
+    @Min(0)
+    public readonly startTime: number;
```

Comment:
You could use `nextRecurrence` instead so when creating an activity, it is clear when it triggers next.